rocketpy/motors/tank_geometry.py

```diff
@@ -369,28 +369,37 @@ def __init__(self, radius, height, spherical_caps=False, geometry_dict=dict()):
         height : float
             Height of the cylindrical tank, in meters.
         spherical_caps : bool, optional
-            If True, the tank will have spherical caps. The default is False.
+            If True, the tank will have spherical caps. The height of the
+            tank is maintained. The default is False.
         geometry_dict : dict, optional
             Dictionary containing the geometry of the tank. See TankGeometry.
         """
         super().__init__(geometry_dict)
         self.has_caps = False
-        self.add_geometry((-height / 2, height / 2), radius)
-        self.add_spherical_caps() if spherical_caps else None
+        if spherical_caps:
+            self.add_geometry((-height / 2 + radius, height / 2 - radius), radius)
+            self.add_spherical_caps(height)
+        else:
+            self.add_geometry((-height / 2, height / 2), radius)
 
-    def add_spherical_caps(self):
+    def add_spherical_caps(self, total_height):
```

Comment:
I would remove this parameter since this value of `total_height` might have any value so that the Tank geometry may be completely reshaped. Therefore I don't think this should be the behavior of a method that simply add spherical caps.

I implemented this method so that someone that made an instance of a CylindricalTank without caps can add standard caps to it should he want. Therefore, in this reasoning, it made sense to change the total height of the tank since its geometry was changed (without any need for the `total_height` parameter).